cron_jobs/region_cron.php

```diff
@@ -0,0 +1,186 @@
+require_once __DIR__ . '/../classes/War/WarManager.php';
+require_once __DIR__ . '/../classes/travel/Patrol.php';
+
+$system = new System();
+$system->db->connect();
+
+if (!$system->db->con) {
+    throw new RuntimeException("Error connecting to DB!");
+}
+
+$player = false;
+if (isset($_SESSION['user_id'])) {
+    $player = User::loadFromId($system, $_SESSION['user_id']);
+    $player->loadData();
+    $production_key = "RuN ON ProDUCtIon";
+
+    if ($player->staff_manager->isHeadAdmin()) {
+        if (isset($_GET['run_script']) && $_GET['run_script'] == 'true') {
+            if ($system->environment === System::ENVIRONMENT_PROD) {
+                $debug = false;
+                if (isset($_GET['debug'])) {
+                    $debug = $system->db->clean($_GET['debug']);
+                }
+                if (!isset($_POST['confirm']) || $_POST['confirm'] !== $production_key) {
+                    echo "WARNING PRODUCTION ENVIRONMENT! Confirm you would like to run this adhoc on <b>PRODUCTION ENVIRONMENT</b>!!!<br />
+                To confirm, type the following (case sensitive): $production_key<br />
+                <form action='{$system->router->base_url}/cron_jobs/region_cron.php?run_script=true' method='post'>
+                    <input type='text' name='confirm' /><input type='submit' value='Run Script' />
+                </form>";
+                } else {
+                    hourlyRegion($system, $debug);
+                    $player->staff_manager->staffLog(
+                        StaffManager::STAFF_LOG_ADMIN,
+                        "{$player->user_name}({$player->user_id}) manually ran region cron."
+                    );
+                }
+            } else {
+                $debug = false;
+                if (isset($_GET['debug'])) {
+                    $debug = $system->db->clean($_GET['debug']);
+                }
+                hourlyRegion($system, $debug);
+                $player->staff_manager->staffLog(
+                    StaffManager::STAFF_LOG_ADMIN,
+                    "{$player->user_name}({$player->user_id}) manually ran region cron."
+                );
+            }
+        } else {
+            echo "You can run the region cron script Adhoc. This is not reversible.<br><a href='{$system->router->base_url}/cron_jobs/region_cron.php?run_script=true'>Run</a><br><a href='{$system->router->base_url}/cron_jobs/region_cron.php?run_script=true&debug=true'>Debug</a>";
+        }
+    }
+} else {
+    // Check for verify to run cron
+    $run_ok = false;
+    if (php_sapi_name() == 'cli') {
+        $run_ok = true;
+    } else if ($_SERVER['REMOTE_ADDR'] == $_SERVER['SERVER_ADDR']) {
+        $run_ok = true;
+    }
+
+    if ($run_ok) {
+        hourlyRegion($system);
+        $system->log('cron', 'Hourly Region', "Regions have been processed.");
+    } else {
+        $system->log('cron', 'Invalid access', "Attempted access by " . $_SERVER['REMOTE_ADDR']);
+    }
+}
```

Comment:
Hmm, starting to feel like we should pull this into some kind of standard cron auth function

Reply:
Agreed, we're going to need some more jobs probably, like monthly tracking (pvp, points, etc), so standardizing would be great.